app/handlers/handlers_services.py

```diff
@@ -70,10 +71,30 @@ def get_ca_cert(tls_secret: kubernetes.client.V1Secret) -> str:
 TLS_OBJECT_ANNOTATION = "resource.twingate.com/tlsSecret"
 
 
+def validate_load_balancer_status(status: Status, service_name: str) -> None:
+    if not (ingress := status.get("loadBalancer", {}).get("ingress")):
+        raise kopf.TemporaryError(
+            f"Kubernetes Service: {service_name} LoadBalancer IP is not ready.",
+            delay=30,
+        )
+
+    if not ingress[0].get("ip"):
+        raise kopf.TemporaryError(
+            f"Kubernetes Service: {service_name} LoadBalancer IP is not ready.",
+            delay=30,
+        )
+
+
+class ServiceType(StrEnum):
+    CLUSTER_IP = "ClusterIP"
+    LOAD_BALANCER = "LoadBalancer"
+
+
 def service_to_twingate_resource(service_body: Body, namespace: str) -> dict:
     meta = service_body.metadata
     spec = service_body.spec
-    service_name = service_body.meta.name
+    status = service_body.status
+    service_name = service_body.meta.name or ""
```

Comment:
This change is to fix mypy `error: Argument 2 to "validate_load_balancer_status" has incompatible type "str | None"; expected "str"  [arg-type]`

Reply:
I would avoid setting a variable value to fix a type issue. Instead I would consider
- Annotate `service_name: str | None` in `validate_load_balancer_status()` helper function
- Use `cast(str, service_body.meta.name)` to let mypy know `name` cannot be `None` here
- Pass the `service_body: Body` to the `validate_load_balancer_status`

Reply:
Lets go with the second option since, `service_name` cannot be `None` here